modules/rte_mis_school/rte_mis_school.module

```diff
@@ -619,14 +619,20 @@ function rte_mis_school_form_level_education_details_validate(array &$form, Form
 
         // Combine the values into a unique string for comparison.
         // Check for conflicting education levels.
-        $key_for_conflict_check = $board_type . '|' . $category . '|' . $medium;
-        if (isset($education_levels_checked[$key_for_conflict_check])) {
-          if (in_array($education_level, $education_levels_checked[$key_for_conflict_check])) {
-            $form_state->setErrorByName("field_education_details][$key][subform][field_education_level", t('Different entries of Education Level are not allowed for the same Board Type, Education Type, and Medium.'));
-            return;
+        $categories_to_check = ($category === 'co-ed') ? ['boys', 'girls'] : [$category];
+        foreach ($categories_to_check as $cat) {
+          $key_for_conflict_check = $board_type . '|' . $cat . '|' . $medium;
+
+          // Check for conflicting education levels for this category.
+          if (isset($education_levels_checked[$key_for_conflict_check])) {
+            if (in_array($education_level, $education_levels_checked[$key_for_conflict_check])) {
```

Comment:
Let's use `&&`  instead of using two if statements.

Reply:
ok